WordPress/Sniffs/WP/OptionAutoloadSniff.php

```diff
@@ -0,0 +1,439 @@
+				$param_second_token = false;
+				$normalized_value   = substr( $normalized_value, 1 );
+			}
+		}
+
+		if ( isset( $this->autoload_is_optional[ $function_name ] ) ) {
+			$valid_values = $this->valid_values_add_and_update;
+		} else {
+			$valid_values = $this->valid_values_other_functions;
+		}
+
+		if ( isset( $valid_values[ $normalized_value ] ) ) {
+			$this->phpcsFile->recordMetric( $param_first_token, self::METRIC_NAME, $normalized_value );
+			return;
+		}
+
+		if ( in_array( $this->tokens[ $param_first_token ]['code'], array( \T_VARIABLE, \T_STRING ), true )
+			&& 'null' !== strtolower( $this->tokens[ $param_first_token ]['content'] )
+		) {
+			// Bail early if the first non-empty token in the parameter is T_VARIABLE or T_STRING as
+			// this means it is not possible to determine the value.
+			$this->phpcsFile->recordMetric( $param_first_token, self::METRIC_NAME, 'undetermined value' );
+			return;
+		}
+
+		if ( $param_second_token
+			&& false === isset( Collections::arrayOpenTokensBC()[ $this->tokens[ $param_first_token ]['code'] ] )
+		) {
+			// Bail early if the parameter has two or more non-empty tokens and the second token is
+			// not an array opener as this means an undetermined param value or a value that is not
+			// easy to determine.
+			$this->phpcsFile->recordMetric( $param_first_token, self::METRIC_NAME, 'undetermined value' );
+			return;
+		}
+
+		$autoload_value = TextStrings::stripQuotes( $autoload_info['clean'] );
+
+		$known_discouraged_values = array_merge( $this->deprecated_values, $this->internal_values_non_fixable, $this->internal_values_fixable );
+
+		if ( isset( $known_discouraged_values[ $autoload_value ] ) ) {
+			$metric_value = $autoload_value;
+		} else {
+			$metric_value = 'other value';
+		}
+
+		$this->phpcsFile->recordMetric( $param_first_token, self::METRIC_NAME, $metric_value );
+
+		if ( isset( $this->deprecated_values[ $autoload_value ] ) ) {
+			$message    = 'The use of `%s` as the value of the `$autoload` parameter is deprecated. Use `%s` instead.';
+			$error_code = 'Deprecated';
+			$data       = array( $autoload_info['clean'], $this->fixable_values[ $autoload_value ] );
+		} elseif ( isset( $this->internal_values_fixable[ $autoload_value ] ) ) {
+			$message    = 'The use of `%s` as the value of the `$autoload` parameter is discouraged. Use `%s` instead.';
+			$error_code = 'InternalUseOnly';
+			$data       = array( $autoload_info['clean'], $this->fixable_values[ $autoload_value ] );
+		} elseif ( isset( $this->internal_values_non_fixable [ $autoload_value ] ) ) {
+			$message    = 'The use of `%s` as the value of the `$autoload` parameter is discouraged.';
+			$error_code = 'InternalUseOnly';
+			$data       = array( $autoload_info['clean'] );
+		} else {
+			$valid_values        = array_map(
+				function ( $value ) {
+					return '`' . $value . '`';
+				},
+				$valid_values
+			);
+			$valid_values_string = implode( ', ', $valid_values );
```

Comment:
For your consideration:

```suggestion
			$valid_values_string = '`' . implode( '`, `', $valid_values ) . '`';
```